sql-plugin/src/main/scala/com/nvidia/spark/ParquetCachedBatchSerializer.scala

```diff
@@ -39,11 +39,9 @@ trait GpuCachedBatchSerializer extends CachedBatchSerializer {
 /**
  * User facing wrapper class that calls into the internal version.
  */
-class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer {
+class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer with Proxy {
```

Comment:
Nit: 
Proxy is deprecated in Scala 2.13: https://www.scala-lang.org/api/2.13.x/scala/Proxy.html
Although we are using 2.12 currently, we'll get a warning in the future after Scala is upgraded to 2.13.
```
<scala.binary.version>2.12</scala.binary.version>
```